src/group_theory/quotient_group.lean

```diff
@@ -412,4 +412,86 @@ monoid_hom.to_mul_equiv
 
 end third_iso_thm
 
+
+section trivial
+
+/-These first few lemmas should maybe go somewhere else, where?-/
+
+lemma quot_triv_eqv_gen_triv {α : Type u} (r : α → α → Prop ) :
+  subsingleton (quot r) ↔ forall a b, eqv_gen r a b :=
+begin
+  split,
+  intro h,
+  intros a b,
+  apply quot.exact,
+  apply subsingleton.elim _ _,
+  exact h,
+  intro h,
+  apply subsingleton.intro,
+  intros a b,
+  have ha:= quot.exists_rep a ,
+  have hb:= quot.exists_rep b ,
+  have h3:= h (classical.some ha)(classical.some hb),
+  have h4:= quot.eqv_gen_sound h3,
+  have h5:= classical.some_spec ha,
+  have h6:= classical.some_spec hb,
+  rw ← h5,
+  rw ← h6,
+  exact h4,
+end
+
+
+lemma eqv_gen_triv_rel_triv {α : Type u} (r : α → α → Prop )
+  (h : eqv_gen r = λ _ _, true) (h2 : equivalence r) : r = λ _ _, true:=
+begin
+  rw ← h,
+  ext,
+  simp_rw relation.eqv_gen_iff_of_equivalence h2,
+end
```

Comment:
I think the missing lemma here is actually
```lean
lemma eqv_gen_eq_of_equivalence (h : equivalence r) : eqv_gen r = r :=
-- funext $ λ _, funext $ λ _, propext $ relation.eqv_gen_iff_of_equivalence 
```
which belongs in the same file as `relation.eqv_gen_iff_of_equivalence`

Reply:
isnt this just ``relation.eqv_gen_iff_of_equivalence ``?

Reply:
this maybe means I shouldn't add this lemma and just unpack it where I need it.

Reply:
No, this is a stronger version of that lemma that can be used to rewrite `eqv_gen r` without applying `ext`.

Reply:
Aha I see, Ok I've put this in the relation.lean file.